### PR TITLE
Add rule to disallow mixins

### PR DESCRIPTION
**Pull request by nlfurniss**
Mixins can cause a lot of headaches, and it would be nice to have a rule that disallowed them.

I left most of the heavy lifting of explaining why mixins can be bad to two reat blog posts; figured I didn't need to reinvent the wheel. Feedback especially appreciate on the doc for this rule: are the code examples easy to understand, etc.

cc: @scalvert

### CODE REVIEW

**Comment by scalvert** [comment]
@rwjblue agree that we definitely want to discourage mixin _usage_. That said, in an effort to stop adding new mixins, this rule seems like a good first step.

**Comment by rwjblue** [comment]
> That said, in an effort to stop adding new mixins, this rule seems like a good first step.

Yep, I hope I wasn't confusing there. I think this rule is good generally speaking, just not "the end" ...

File: tests/lib/rules/no-mixins.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+eslintTester.run('no-mixins', rule, {
+  valid: [
+    {
+      code: `
+        import mixin from "some/addon";
+        export default mixin;
+      `,
+    },
+    {
+      code: 'export default mixin.extend({actions: {},});',
```
[the remark this answers is not in the document]
Comment (nlfurniss) [reply tests/lib/rules/no-mixins.js]: Yup. brain fart

**Comment by nlfurniss** [comment]
Renamed rule to `no-new-mixins`

**Comment by kanongil** [comment]
I see that this rule has been enabled by default. Any migration paths for my use case, where I use mixins to add optional functionality to a base component?

**Comment by mehulkar** [comment]
Disable the rule for existing mixins and don’t write new ones?

**Pull request merged**